### ISSUE TITLE
Parcel2 on arm/pi

### ISSUE THREAD

**Issue by koellcode** [parcel-bundler/source-map#33]
# 🐛 bug report

after succesful installing on arm i get a undefined symbol error in sourcemap.node

## 🎛 Configuration (.babelrc, package.json, cli command)

just demo

 cat package.json 
{
  "name": "parcel2-demo",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "dependencies": {},
  "scripts": {
    "start": "parcel serve ./src/index.html",
    "build": "parcel build ./src/index.html"
  },
  "devDependencies": {
    "parcel": "^2.0.0-beta.1"
  }
}

## 🤔 Expected Behavior

should build

## 😯 Current Behavior

doesn‘t build, fails with: 

$ parcel build ./src/index.html
internal/modules/cjs/loader.js:1127
  return process.dlopen(module, path.toNamespacedPath(filename));
                 ^

Error: /home/thomas/git/parcel2-demo/node_modules/@parcel/source-map/build/Release/sourcemap.node: undefined symbol: _ZN4Napi7details31needs_objectwrap_destructor_fixE
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:1127:18)
    at Module.load (internal/modules/cjs/loader.js:941:32)
    at Function.Module._load (internal/modules/cjs/loader.js:782:14)
    at Module.require (internal/modules/cjs/loader.js:965:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at load (/home/thomas/git/parcel2-demo/node_modules/node-gyp-build/index.js:21:10)
    at Object.<anonymous> (/home/thomas/git/parcel2-demo/node_modules/@parcel/source-map/dist/node.js:14:43)
    at Module._compile (internal/modules/cjs/loader.js:1076:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1097:10)
    at Module.load (internal/modules/cjs/loader.js:941:32)
error Command failed with exit code 1.

## 🔦 Context

i run code-server on a raspi4 and try some PoCs there sometimes, this time i wanted to try out parcel2
## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0 beta 1
| Node             | 14.11.0
| npm/Yarn         | yarn 1.22.5
| Operating System | Arch Linux ARM

**Comment by WA9ACE** [comment]
This is also affecting M1 Macs.

**Comment by DeMoorJasper** [comment]
@WA9ACE that's not good, unfortunately nobody on the Parcel core team currently has an M1 mac and Azure doesn't support it yet either... so we can't really test this at the moment.

I do have one on order but it won't be delivered until somewhere mid-december, so until than I can't really debug and fix this.

**Comment by mischnic** [comment]
Just to clarify: you're not using rosetta (but an actual ARM Node binary)?

**Comment by DeMoorJasper** [comment]
Please continue the Mac M1 conversation in #41

**Comment by mischnic** [comment]
I don't get an error on an Raspberry Pi 1 B (armv6l) and `parcel@nightly` (with both serve and build), it's just unbearably slow 😄

**Comment by nathanielparry78** [comment]
Individually installing `@parcel/source-map` fixed this for me when running parcel@2.0.0-beta.1.

What I discovered was `parcel@2.0.0-beta.1` uses an older version of `@parcel/source-map` that is running `node-addon-api@2.X.X` NOT v3 as indicated above. When I ran `npm i @parcel/source-map` it installed `@parcel/source-map@2.0.0-alpha.4.20` which DOES have v3 of `node-addon-api`.

Hope this helps somebody.

**Comment by mischnic** [comment]
So I guess we can close this because it's fixed when the latest version?

**Comment by nathanielparry78** [comment]
It is included in the nightly version, but not the beta version.

Maybe we could get `parcel@2.0.0-beta.2` that includes this fix? :D

**Comment by Javarome** [comment]
Hello, I'm running into this problem too on a MacBook M1, but I fail to workaround it using your tip @nathanielparry78

I used:
```
  "devDependencies": {
    "@parcel/source-map": "^2.0.0-alpha.4.20",      // Your tip @nathanielparry78
    "@parcel/transformer-sass": "^2.0.0-alpha.3",   // I need this one, which prevents me to use parcel beta2 or nightly
    "@types/jest": "26.0.20",
    "identity-obj-proxy": "^3.0.0",
    "jest": "26.6.3",
    "parcel": "2.0.0-beta.1",
    "sass": "1.32.8"
  }
  ```
  Is there something I'm missing?

**Comment by DeMoorJasper** [comment]
Resolved by #52
